### PR TITLE
New scorer

### PR DESCRIPTION
**Pull request by yramachers**
Branch 'newScorer' has the example RE01 as origin and seems to work fine. Caution - currently simply prints trajectories to screen without(!) filterting, every little e- is there - very long output. All that information needs filtering and then output to disk, easiest with the existing root TTree output mechanism - accessible at the same point, i.e. EndOfEventAction function in WLGDEvent.

See comment in issue for more discussion of this PR and the other branch 'simplerScorer'. I would still like to explore the difference or connections betwen hits in detectors and trajectories from tracks, if any. At the moment they appear to be complementary pieces of information which would be fine as they come together at the end of event, but there is nice and simple filtering for hits but nothing at the moment for the trajectories, not all of which are of any interest. More to be done there.

### CODE REVIEW

**Comment by yramachers** [comment]
Slightly more realistic extension of scoring - track id and parent id for hits and the same for trajectories in event action; all written to root file - to be expanded to store all the relevant trajectory data on branches. Just those ID's for now.
Easier to run since output print is filtered on particle name (and Z, A, for an ion using particle definition object) just like the SDFilter but here with a simple if clause.

File: src/WLGDTrajectory.cc [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+  fTrackID            = aTrack->GetTrackID();
+  fParentID           = aTrack->GetParentID();
+
+  fPositionRecord = new WLGDTrajectoryPointContainer();
+  fPositionRecord->push_back(new G4TrajectoryPoint(aTrack->GetPosition()));
+
+  fMomentum       = aTrack->GetMomentum();
+  fVertexPosition = aTrack->GetVertexPosition();
+  fGlobalTime     = aTrack->GetGlobalTime();
+}
```
Comment (drbenmorgan) [inline src/WLGDTrajectory.cc:32]:
```suggestion
: G4VTrajectory(),
  fPositionRecord{new WLGDTrajectoryPointContainer()},
  fTrackID{aTrack->GetTrackID()},
  fParentID{aTrack->GetParentID()},
  fParticleDefinition{aTrack->GetDefinition()},
  fParticleName{fParticleDefinition->GetParticleName()},
  fPDGCharge{fParticleDefinition->GetPDGCharge()},
  fPDGEncoding{fParticleDefinition->GetPDGEncoding()},
  fMomentum{aTrack->GetMomentum()},
  fVertexPosition{aTrack->GetVertexPosition()},
  fGlobalTime{aTrack->GetGlobalTime()}
{
  fPositionRecord->push_back(new G4TrajectoryPoint(aTrack->GetPosition()));
}
```

File: src/WLGDStackingAction.cc [hunk 2]
```diff
@@ -0,0 +1,26 @@
+#include "WLGDStackingAction.hh"
+#include "WLGDTrackInformation.hh"
+
+WLGDStackingAction::WLGDStackingAction()
+: G4UserStackingAction()
+{
+  ;
+}
+
+WLGDStackingAction::~WLGDStackingAction() { ; }
```
[the remark this answers is not in the document]
Comment (yramachers) [reply src/WLGDStackingAction.cc:10]: next commit

**Comment by yramachers** [comment]
I am putting together the trivial feature enhancements (new scorers) at the moment and would commit these here. Just adding the required primitive scorers some of which exist already on the other branch. That way, all the required scoring is done with this commit and other features can go on subsequent pull requests. Should all be done today.

**Comment by yramachers** [comment]
Trajectory data is now also saved: PDG, Vertex position, NEntries and Points. The sensitive detector is set to only Ge while trajectories filter on all neutrons and Ge77, could add muon trajectories here - will mostly be just one trajectory extra. Text output is also reduced (no ShowTrajectory anymore). Starts to require a post-processing root macro - plenty of info in the root file. Messengers next to make all commands including quite a few verbose commands.

**Comment by drbenmorgan** [comment]
Thanks @yramachers, just post a comment when your commits are complete and we should be good to merge barring any compile issues to address!

**Comment by yramachers** [comment]
Yes, commits on this branch should be finished. Compiles and runs fine on SCRTP. Good to merge once the check passes.

**Pull request merged**